### ISSUE TITLE
Support for AnimationTimer

### ISSUE THREAD

**Issue by danielkroeni** [Frege/FregeFX#7]
In order to use [`javafx.animation.AnimationTimer`](https://docs.oracle.com/javase/8/javafx/api/javafx/animation/AnimationTimer.html) you have to subclass it and override its handle method. This is not an option if you want to use it directly from frege. So I implemented a 'builder' where you can just pass the function you want to be called on every frame:

``` haskell
data AnimationTimerBuilder = mutable native graphics.AnimationTimerBuilder where
   native from graphics.AnimationTimerBuilder.from :: (Long -> IO ()) -> IO AnimationTimer
```

``` java
import javafx.animation.AnimationTimer;
import frege.prelude.PreludeBase;
import frege.runtime.Lambda;

public class AnimationTimerBuilder {
  public static AnimationTimer from(Lambda handler) {
    return new AnimationTimer() {
      @Override
      public void handle(long arg0) {
        frege.runtime.Delayed.<java.lang.Void>forced(
          PreludeBase.TST.performUnsafe(handler.apply(arg0).result().forced())
        );
      }
    };
  }
}
```

Now you can use it like this:

``` haskell
do animTimer <- AnimationTimerBuilder.from $ \time -> do
  println "Hi" 
  println time

animTimer.start
```

If you could add something similar to your fx utils module, I could drop my version.

**Comment by Dierk** [comment]
Cool, thanks!

My goal was to build sth like https://github.com/Dierk/groovyfx/blob/master/groovyfx/src/demo/groovy/AnimationDemo.groovy

But so we have sth to start with. 

Thanks a lot
Dierk

sent from:mobile

**Comment by Ingo60** [comment]
I am wondering why this works. Probably because the forced value is not used and hence no cast is actually done.

Certainly, the cast to `java.lang.Void` doesn't feel right to me. I would've choosen `Object`, probably. 
In fact, an IO () is under the hood a method that returns (), and the type () is an enumeration, which is currently implemented as a short that indicates the constructor value. Since () is the only constrctor of (), it should always be 0.

**Comment by danielkroeni** [comment]
You are right. No idea what I was thinking. `Void` is definitely the wrong choice.
